Compiler errors (and warnings) on GCC 11.2

I got this from running `pip wheel runtime/` on a Debian machine that has GCC 11.2 by default.

```
  FAILED: runtime/src/iree/base/CMakeFiles/iree_base_base.objects.dir/status.c.o
  /usr/bin/cc  -isystem /usr/local/google/home/laurenzo/src/iree -isystem /usr/local/google/home/laurenzo/src/iree/runtime/build/cmake_build -isystem /usr/local/google/home/laurenzo/src/iree/runtime/src -isystem /usr/local/google/home/laurenzo/src/iree/runtime/build/cmake_build/runtime/src -O3 -DNDEBUG -fPIC -fvisibility=hidden -Wall -Wno-address-of-packed-member -Wno-comment -Wno-format-zero-length -Wno-pointer-sign -Wno-sign-compare -Wno-unused-function -std=gnu11 -MD -MT runtime/src/iree/base/CMakeFiles/iree_base_base.objects.dir/status.c.o -MF runtime/src/iree/base/CMakeFiles/iree_base_base.objects.dir/status.c.o.d -o runtime/src/iree/base/CMakeFiles/iree_base_base.objects.dir/status.c.o -c /usr/local/google/home/laurenzo/src/iree/runtime/src/iree/base/status.c
  /usr/local/google/home/laurenzo/src/iree/runtime/src/iree/base/status.c:631:1: warning: ‘nodiscard’ attribute ignored [-Wattributes]
    631 | static IREE_MUST_USE_RESULT iree_status_t
        | ^~~~~~
  /usr/local/google/home/laurenzo/src/iree/runtime/src/iree/base/status.c:631:29: error: expected identifier or ‘(’ before ‘iree_status_t’
    631 | static IREE_MUST_USE_RESULT iree_status_t
        |                             ^~~~~~~~~~~~~
  /usr/local/google/home/laurenzo/src/iree/runtime/src/iree/base/status.c: In function ‘iree_status_annotate_f’:
  /usr/local/google/home/laurenzo/src/iree/runtime/src/iree/base/status.c:687:7: warning: implicit declaration of function ‘iree_status_annotate_vf’; did you mean ‘iree_status_annotate_f’? [-Wimplicit-function-declaration]
    687 |       iree_status_annotate_vf(base_status, format, varargs_0, varargs_1);
        |       ^~~~~~~~~~~~~~~~~~~~~~~
        |       iree_status_annotate_f
  /usr/local/google/home/laurenzo/src/iree/runtime/src/iree/base/status.c:687:7: warning: initialization of ‘iree_status_t’ {aka ‘struct iree_status_handle_t *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
```

--- a/runtime/src/iree/base/status.c
+++ b/runtime/src/iree/base/status.c
@@ -628,9 +628,9 @@ iree_status_annotate(iree_status_t base_status, iree_string_view_t message) {
                                     (iree_status_payload_t*)payload);
 }
 
-static IREE_MUST_USE_RESULT iree_status_t
-iree_status_annotate_vf(iree_status_t base_status, const char* format,
-                        va_list varargs_0, va_list varargs_1) {
+IREE_MUST_USE_RESULT static iree_status_t iree_status_annotate_vf(
+    iree_status_t base_status, const char* format, va_list varargs_0,
+    va_list varargs_1) {
   if (iree_status_is_ok(base_status)) return base_status;
 
   // If there's no storage yet we can just reuse normal allocation. Both that

